File: python_roguelike/data/combatant/enemy_data.py
from dataclasses import dataclass, field
from typing import List
from .combatant_data import CombatantData
from ..weighted_choice import WeightedChoice
from ..combat_action_data import CombatActionData


@dataclass
class EnemyData(CombatantData):
    star_rating: int = 1
    is_boss: bool = False
    action_set: List[WeightedChoice] = field(default_factory=list)
    special_ability_cooldown: int = 1

File: src/Roguelike.Core/Core/Data/Combatant/EnemyData.cs


using System.Collections.Generic;

namespace Roguelike.Data
{


    public class EnemyData : CombatantData
    {


        public int StarRating { get; set; }


        public bool IsBoss { get; set; } = false;


        public List<WeightedChoice<CombatActionData>> ActionSet { get; set; } = new List<WeightedChoice<CombatActionData>>();


        public int SpecialAbilityCooldown { get; set; } = 1;
    }
}
